spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/PubSubTopicUtils.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.support;
+
+import com.google.pubsub.v1.ProjectTopicName;
+
+import org.springframework.util.Assert;
+
+/**
+ * Various utility methods for dealing with Pub/Sub topics.
+ *
+ * @author Mike Eltsufin
+ * @since 1.2
+ */
+public final class PubSubTopicUtils {
+
+	private PubSubTopicUtils() {
+	}
+
+	/**
+	 * Creates a {@link ProjectTopicName} based on a topic name within a project or the
+	 * fully-qualified topic name. If the specified topic is in the
+	 * projects/&lt;project_name&gt;/topics/&lt;topic_name&gt; format, then the {@code projectId} is
+	 * ignored}
+	 * @param projectId the project ID to use if the topic is not a fully-qualified name
+	 * @param topic the topic name in the project or the fully-qualified project name
+	 * @return the Pub/Sub object representing the topic name
+	 */
+	public static ProjectTopicName parseTopic(String projectId, String topic) {
```

Comment:
I would change the order since it looks like `projectId` is optional. We might name it as a `defaultProjectId`.
Also would be great to mark it with the `@org.springframework.lang.Nullable`